code/game/mecha/equipment/tools/janitor_tools.dm

```diff
@@ -0,0 +1,377 @@
+	create_reagents(1000)
+	reagents.add_reagent("water", 1000)
+	START_PROCESSING(SSobj, src)
+	..()
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/emag_act(mob/user)
+	. = ..()
+	emagged = TRUE
+	to_chat(user, "<span class='notice'>You short out the automatic watering system on [src].</span>")
+	reagents.clear_reagents()
+	refill_reagent = "lube"
+	refill_cost = 50
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/action(atom/target)
+	if(istype(target, /obj/structure/reagent_dispensers/watertank) && get_dist(chassis,target) <= 1)
+		var/obj/structure/reagent_dispensers/watertank/WT = target
+		WT.reagents.trans_to(src, 1000)
+		occupant_message("<span class='notice'>Mop refilled.</span>")
+		playsound(chassis, 'sound/effects/refill.ogg', 50, TRUE, -6)
+	else
+		if(get_dist(chassis, target) > 2)
+			return
+		if(reagents.total_volume > 0)
+			if(world.time > mop_sound_cooldown)
+				playsound(loc, pick('sound/weapons/mopping1.ogg', 'sound/weapons/mopping2.ogg'), 30, TRUE, -1)
+				mop_sound_cooldown = world.time + MOP_SOUND_CD
+			// 3x3 mopping area
+			var/turf/target_turf = get_turf(target)
+			if(!istype(target_turf) || iswallturf(target_turf))
+				return
+			for(var/turf/current_target_turf in view(1, target))
+				INVOKE_ASYNC(current_target_turf, TYPE_PROC_REF(/atom, cleaning_act), chassis.occupant, src, mop_speed, "mop", ".")
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/post_clean(atom/target, mob/user)
+	var/turf/T = get_turf(target)
+	if(issimulatedturf(T))
+		reagents.reaction(T, REAGENT_TOUCH, 10)	// 10 is the multiplier for the reaction effect. 10 is needed to properly wet a floor.
+	reagents.remove_any(1)	// reaction() doesn't use up the reagents
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/can_clean()
+	if(reagents.has_reagent("water", 1) || reagents.has_reagent("cleaner", 1) || reagents.has_reagent("holywater", 1))
+		return TRUE
+	else
+		return FALSE
+
+// Auto-regeneration of water. Takes energy.
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/process()
+	if(reagents.total_volume < 1000)
+		reagents.add_reagent(refill_reagent, refill_rate)
+		chassis.use_power(refill_cost)
+		update_equip_info()
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/get_equip_info()
+	var/output = ..()
+	if(output)
+		return "[output] \[<a href='byond://?src=[UID()];toggle_mode=1'>Refill [refill_enabled? "Enabled" : "Disabled"]</a>\] \[[src.reagents.total_volume]\]"
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/Topic(href,href_list)
+	..()
+	var/datum/topic_input/afilter = new (href,href_list)
+	if(afilter.get("toggle_mode"))
+		refill_enabled = !refill_enabled
+		if(refill_enabled)
+			START_PROCESSING(SSobj, src)
+		else
+			STOP_PROCESSING(SSobj, src)
+		update_equip_info()
+		return
+
+#undef MOP_SOUND_CD
+
+// Light Replacer
+
+#define LIGHT_OK 0
+#define LIGHT_EMPTY 1
+#define LIGHT_BROKEN 2
+#define LIGHT_BURNED 3
```

Comment:
Move these to a file in code/__DEFINES. We really don't need them defined in three different files.